### PR TITLE
MTR-497 don't switch profiles if the lock doesn't support tamper

### PR DESCRIPTION
**Pull request by greens**

### CODE REVIEW

File: drivers/SmartThings/matter-lock/src/init.lua [hunk 1]
```diff
@@ -506,7 +506,7 @@ local function device_added(driver, device)
   --Note: May want to write RequirePINForRemoteOperation, to avoid cota cases if possible.
   device:emit_event(capabilities.tamperAlert.tamper.clear())
   local eps = device:get_endpoints(DoorLock.ID, {feature_bitmap = DoorLock.types.DoorLockFeature.PIN_CREDENTIALS})
-  if #eps == 0 then
+  if #eps == 0 and device:supports_capability_by_id(capabilities.tamperAlert.ID) then
```
Comment (tpmanley) [inline drivers/SmartThings/matter-lock/src/init.lua:509]: This would hit the else case if the device doesn't support PIN creds or tamper

**Review by cjswedes — APPROVED** [review]

File: drivers/SmartThings/matter-lock/src/init.lua [hunk 2]
```diff
@@ -507,8 +507,10 @@ local function device_added(driver, device)
   device:emit_event(capabilities.tamperAlert.tamper.clear())
   local eps = device:get_endpoints(DoorLock.ID, {feature_bitmap = DoorLock.types.DoorLockFeature.PIN_CREDENTIALS})
   if #eps == 0 then
-    device.log.debug("Device does not support lockCodes")
-    device:try_update_metadata({profile = "lock-without-codes"})
+    if device:supports_capability_by_id(capabilities.tamperAlert.ID) then
+      device.log.debug("Device does not support lockCodes")
+      device:try_update_metadata({profile = "lock-without-codes"})
+    end
```
Comment (tpmanley) [inline drivers/SmartThings/matter-lock/src/init.lua:513]: What do you think about adding an else with a warning that we don't have a profile for a device that doesn't support codes or tamper?

**Review by tpmanley — APPROVED** [review]

File: drivers/SmartThings/matter-lock/src/init.lua (hunk 2 above)
Comment (greens) [reply drivers/SmartThings/matter-lock/src/init.lua:513]: I think this is only temporary until the incoming profile gets merged. Then we can actually do switching in that `else`

Comment (greens) [reply drivers/SmartThings/matter-lock/src/init.lua:515]: updated anyway, though

**Pull request merged**